sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -454,8 +455,8 @@ case class Lower(child: Expression)
   override protected def withNewChildInternal(newChild: Expression): Lower = copy(child = newChild)
 }
 
-/** A base trait for functions that compare two strings, returning a boolean. */
-abstract class StringPredicate extends BinaryExpression
+/** A base trait for functions that compare two strings or binaries, returning a boolean. */
+abstract class StringBinaryPredicate extends BinaryExpression
```

Comment:
do we need to rename this? The new `BinaryPredicate` does not extend it.

Reply:
> do we need to rename this? The new `BinaryPredicate` does not extend it.

Double checked, should rename back. done.